src/plots/smith/layout_attributes.js

```diff
@@ -0,0 +1,121 @@
+'use strict';
+
+var colorAttrs = require('../../components/color/attributes');
+var axesAttrs = require('../cartesian/layout_attributes');
+var domainAttrs = require('../domain').attributes;
+var extendFlat = require('../../lib').extendFlat;
+var overrideAll = require('../../plot_api/edit_types').overrideAll;
+
+var axisLineGridAttr = overrideAll({
+    color: axesAttrs.color,
+    showline: extendFlat({}, axesAttrs.showline, {dflt: true}),
+    linecolor: axesAttrs.linecolor,
+    linewidth: axesAttrs.linewidth,
+    showgrid: extendFlat({}, axesAttrs.showgrid, {dflt: true}),
+    gridcolor: axesAttrs.gridcolor,
+    gridwidth: axesAttrs.gridwidth
+}, 'plot', 'from-root');
+
+var axisTickAttrs = overrideAll({
+    ticklen: axesAttrs.ticklen,
+    tickwidth: extendFlat({}, axesAttrs.tickwidth, {dflt: 2}),
+    tickcolor: axesAttrs.tickcolor,
+    showticklabels: axesAttrs.showticklabels,
+    showtickprefix: axesAttrs.showtickprefix,
+    tickprefix: axesAttrs.tickprefix,
+    showticksuffix: axesAttrs.showticksuffix,
+    ticksuffix: axesAttrs.ticksuffix,
+    tickfont: axesAttrs.tickfont,
+    tickformat: axesAttrs.tickformat,
+    hoverformat: axesAttrs.hoverformat,
+    layer: axesAttrs.layer
+}, 'plot', 'from-root');
+
+var realAxisAttrs = {
+    visible: extendFlat({}, axesAttrs.visible, {dflt: true}),
+
+    tickvals: {
+        dflt: [
+            0.2, 0.4, 0.6, 0.8,
+            1, 1.5, 2, 3, 4, 5, 10, 20
+        ],
+        valType: 'data_array',
+        editType: 'plot',
+        description: 'Sets the values at which ticks on this axis appear.'
+    },
+
+    tickangle: extendFlat({}, axesAttrs.tickangle, {dflt: 90}),
+
+    ticks: {
+        valType: 'enumerated',
+        values: ['top', 'bottom', ''],
```

Comment:
Unlike `polar.radialaxis` that has rather confusing `outside` and `inside` options; I decided not to use those and instead provide `top`, `bottom` options.